### PR TITLE
[KernelGen][Nvidia] Add expand_copy operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `expand_copy`. Returns a new tensor that is a copy of the input expanded to the specified size, broadcasting singleton dimensions.

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (15 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Performance
Test command: `pytest benchmark/test_expand_copy.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.121 | 1.776 | 0.631 |
| 64, 64 | 0.006 | 0.006 | 0.968 |
| 4096, 4096 | 0.024 | 0.054 | 0.441 |
| 64, 512, 512 | 0.024 | 0.170 | 0.142 |
| 1024, 1024, 1024 | 1.148 | 10.120 | 0.113 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 2.264 | 2.494 | 0.908 |
| 64, 64 | 0.006 | 0.006 | 0.925 |
| 4096, 4096 | 0.041 | 0.058 | 0.706 |
| 64, 512, 512 | 0.042 | 0.170 | 0.246 |
| 1024, 1024, 1024 | 2.263 | 10.159 | 0.223 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.152 | 1.775 | 0.649 |
| 64, 64 | 0.005 | 0.005 | 0.994 |
| 4096, 4096 | 0.024 | 0.053 | 0.451 |
| 64, 512, 512 | 0.024 | 0.170 | 0.142 |
| 1024, 1024, 1024 | 1.160 | 10.120 | 0.115 |

**Arithmetic Mean Speedup: 0.510x**
## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 0.510 | Primary |
| Tianshu | PASS | PASS (42 cases) | 0.83 | — |
| Muxi | PASS | PASS (42 cases) | 1.23 | — |
| Ascend | PASS | — | — | — |
| Hygon | PASS | PASS (42 cases) | 1.16 | — |

## Files Changed
- `src/flag_gems/ops/expand_copy.py`: Triton kernel implementation
- `tests/test_expand_copy.py`: Accuracy test
- `benchmark/test_expand_copy.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.4)

### CODE REVIEW

**Comment by tengqm** [comment]
performance number for larger data set is not good, try optimize?

File: src/flag_gems/ops/expand_copy.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import torch
+
+from flag_gems.ops.copy import copy_
+
+logger = logging.getLogger(__name__)
+
+
+def expand_copy(x: torch.Tensor, size) -> torch.Tensor:
+    logger.debug("GEMS EXPAND_COPY")
```
Comment (tengqm) [inline src/flag_gems/ops/expand_copy.py]: This line should be after the docstring lines.

File: src/flag_gems/ops/expand_copy.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    # Use torch.expand to get a broadcasted view with correct strides
+    expanded_src = x.expand(size_tuple)
+
+    # Ensure expanded_src is on the right device (expand preserves device)
+    if expanded_src.device != device:
+        expanded_src = expanded_src.to(device)
+
+    # Use FlagGems copy_ which properly handles broadcasting
+    return copy_(out, expanded_src)
```
Comment (tengqm) [inline src/flag_gems/ops/expand_copy.py:53]: Looks like this is not a Triton operator?

File: src/flag_gems/ops/expand_copy.py (hunk 1 above)
Comment (bwbwzzz) [reply src/flag_gems/ops/expand_copy.py]: thanks，have reviewed！

**Comment by bin913** [comment]
@bwbwzzz Please fix failing check

File: conf/operators.yaml [hunk 3]
```diff
@@ -2492,6 +2492,17 @@ ops:
       - Tensor
     stages:
       - alpha: '5.4'
+  - id: expand_copy
+    description: Returns a new tensor that is a copy of the input expanded to the specified size, broadcasting singleton dimensions.
```
Comment (bin913) [inline conf/operators.yaml]: **Suggestion**: Line exceeds 120 characters (132 chars)

**Recommended fix:**
Consider breaking the line for readability.

<sub>Rule: `CODE_QUALITY`</sub>

File: src/flag_gems/ops/expand_copy.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+logger = logging.getLogger(__name__)
+
+
+def expand_copy(x: torch.Tensor, size) -> torch.Tensor:
+    """
+    Wrapper for aten::expand_copy
+    Creates and returns a copy of x with the given size, broadcasting as needed.
+    """
+    logger.debug("GEMS EXPAND_COPY")
+
```
Comment (bin913) [inline src/flag_gems/ops/expand_copy.py]: **Suggestion**: Trailing whitespace on added line

**Recommended fix:**
Remove trailing whitespace.

<sub>Rule: `CODE_QUALITY`</sub>

**Review by bin913 — COMMENTED** [review]
## FlagGems Automated Review

**Operator:** `expand_copy`
**Verdict:** COMMENT

### Summary
- Errors: 0
- Warnings: 2
- Suggestions: 0

---
<sub>Automated review by FlagGems PR Review Skill</sub>

File: conf/operators.yaml (hunk 3 above)
Comment (bwbwzzz) [inline conf/operators.yaml]:
```suggestion
    description: Returns a copy of the input tensor expanded by broadcasting singleton dimensions.
```

**Comment by w1120029931-bit** [comment]
Add copyright to the new file, please

**Comment by bwbwzzz** [comment]
> Add copyright to the new file, please

i have added it thanks

**Review by bin913 — APPROVED** [review]
